File: C++/A/404A-valeraandx.cpp
#include <bits/stdc++.h>
using namespace std;

class Solver
{
    int board_size;
    vector<string> board;

public:
    void input()
    {
        cin >> board_size;
        for (int i = 0; i < board_size; ++i)
        {
            string x;
            cin >> x;
            board.push_back(x);
        }
    }
    bool isDiagonal()
    {
        if (board.empty() || board.size() != board[0].size())
            return false;
        char diagonalChar = board[0][0];
        for (int i = 0; i < board_size; ++i) // Diagonal
            if (board[i][i] != diagonalChar)
                return false;
        for (int i = 0; i < board_size; ++i) // Anti-diagonal
            if (board[i][board_size - 1 - i] != diagonalChar)
                return false;
        for (int i = 0; i < board_size; ++i) // Other characters
            for (int j = 0; j < board_size; ++j)
                if (((i != j && i + j != board_size - 1) && (board[i][j] != board[0][1] || board[i][j] == diagonalChar)))
                    return false;
        return true;
    }
};

int main()
{
    Solver object;
    object.input();
    cout << ((object.isDiagonal()) ? "YES" : "NO");
}

File: Python/404A-valeraandx.py
class Solver:
    def __init__(self):
        self.board_size = 0
        self.board = []

    def input_board(self):
        self.board_size = int(input())
        for _ in range(self.board_size):
            x = input()
            self.board.append(x)

    def is_diagonal(self):
        if not self.board or len(self.board) != len(self.board[0]):
            return False
        diagonal_char = self.board[0][0]
        for i in range(self.board_size):
            if self.board[i][i] != diagonal_char:
                return False
        for i in range(self.board_size):
            if self.board[i][self.board_size - 1 - i] != diagonal_char:
                return False
        for i in range(self.board_size):
            for j in range(self.board_size):
                if (i != j and i + j != self.board_size - 1 and
                        (self.board[i][j] != self.board[0][1] or self.board[i][j] == diagonal_char)):
                    return False
        return True


obj = Solver()
obj.input_board()
print("YES" if obj.is_diagonal() else "NO")
